cropper/app/lib/Crops.scala

```diff
@@ -60,24 +61,25 @@ object Crops {
     else
       Config.landscapeCropSizingWidths.filter(_ <= bounds.width).map(w => Dimensions(w, math.round(w / aspectRatio)))
 
-  def isInvalidCrop(asset: Asset, specification: CropSource): Boolean = {
-    asset.dimensions.map { dimensions =>
-      val bounds = specification.bounds
+  def isWithinImage(asset: Asset, specification: CropSource): Boolean = {
+    val bounds = specification.bounds
 
-      val outOfBounds = bounds.x > dimensions.width || bounds.y > dimensions.height
-      val cropTooSmall = bounds.width < 1 || bounds.height < 1
-      val cropTooBig = bounds.width > dimensions.width || bounds.height > dimensions.height
+    asset.dimensions.exists { dimensions =>
+      val positiveCoords       = List(bounds.x,     bounds.y     ).forall(_ >= 0)
+      val strictlyPositiveSize = List(bounds.width, bounds.height).forall(_  > 0)
+      val withinBounds = (bounds.x + bounds.width  <= dimensions.width ) &&
+                         (bounds.y + bounds.height <= dimensions.height)
 
-      outOfBounds || cropTooSmall || cropTooBig
-    } getOrElse true
+      positiveCoords && strictlyPositiveSize && withinBounds
```

Comment:
Inverted the logic and made the check more exhaustive (previously, the crop bounds could overflow the actual image dimensions and still be considered valid).